edb/common/ast/base.py

```diff
@@ -192,30 +192,9 @@ def __init__(self, **kwargs):
                 f'cannot instantiate abstract AST node '
                 f'{self.__class__.__name__!r}')
 
-        object.__setattr__(self, 'parent', None)
+        object.__setattr__(self, 'parent', kwargs.get('parent', None))
```

Comment:
I don't think this line is needed, or?..

Reply:
The attribute has to exist due to checks in multiple places and without the setter one caller would have to be changed.

Reply:
What caller?